File: backend/functions/lib/response.py
from typing import Any, Dict, Optional

def create_response(status_code: int, body: Dict[str, Any], headers: Optional[Dict[str, str]] = None) -> Dict[str, Any]:
    """Create a standardized API response"""
    response = {
        'statusCode': status_code,
        'headers': {
            'Content-Type': 'application/json',
            'Access-Control-Allow-Origin': '*',
            'Access-Control-Allow-Credentials': True,
            **(headers or {})
        },
        'body': body
    }
    return response

def success_response(data: Any = None, message: str = 'Success') -> Dict[str, Any]:
    """Create a success response"""
    body = {
        'success': True,
        'message': message
    }
    if data is not None:
        body['data'] = data
    return create_response(200, body)

def error_response(message: str, status_code: int = 400, error_code: Optional[str] = None) -> Dict[str, Any]:
    """Create an error response"""
    body = {
        'success': False,
        'message': message
    }
    if error_code:
        body['error_code'] = error_code
    return create_response(status_code, body)

def validation_error_response(errors: Dict[str, Any]) -> Dict[str, Any]:
    """Create a validation error response"""
    return error_response('Validation error', 400, 'VALIDATION_ERROR')

def unauthorized_response(message: str = 'Unauthorized') -> Dict[str, Any]:
    """Create an unauthorized response"""
    return error_response(message, 401, 'UNAUTHORIZED')

def forbidden_response(message: str = 'Forbidden') -> Dict[str, Any]:
    """Create a forbidden response"""
    return error_response(message, 403, 'FORBIDDEN')

def not_found_response(message: str = 'Resource not found') -> Dict[str, Any]:
    """Create a not found response"""
    return error_response(message, 404, 'NOT_FOUND')

File: backend/lambda/go/shared/api/response.go
package api

import (
	"encoding/json"

	"github.com/aws/aws-lambda-go/events"
)

// APIResponse is the standard response format for all API endpoints
// @Description Standard response format for all API endpoints
type APIResponse struct {
	// Indicates if the operation was successful
	Success bool `json:"success" example:"true"`
	// Human-readable message about the result
	Message string `json:"message" example:"Operation completed successfully"`
	// Response data (if any)
	Data interface{} `json:"data,omitempty"`
	// Error message (if any)
	Error string `json:"error,omitempty" example:"Invalid input provided"`
}

// SendResponse creates a standardized API Gateway HTTP response
func SendResponse(statusCode int, success bool, message string, data interface{}, err string) events.APIGatewayV2HTTPResponse {
	response := APIResponse{
		Success: success,
		Message: message,
		Data:    data,
		Error:   err,
	}
	jsonResponse, _ := json.Marshal(response)
	return events.APIGatewayV2HTTPResponse{
		StatusCode: statusCode,
		Body:       string(jsonResponse),
		Headers: map[string]string{
			"Content-Type":                 "application/json",
			"Access-Control-Allow-Origin":  "*", // Enable CORS
			"Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
			"Access-Control-Allow-Headers": "Content-Type, Authorization",
		},
	}
}

// SendError is a convenience function for sending error responses
func SendError(statusCode int, message string, err string) events.APIGatewayV2HTTPResponse {
	return SendResponse(statusCode, false, message, nil, err)
}

// SendSuccess is a convenience function for sending success responses
func SendSuccess(statusCode int, message string, data interface{}) events.APIGatewayV2HTTPResponse {
	return SendResponse(statusCode, true, message, data, "")
}
